build2cmake/src/config/v2.rs

```diff
@@ -0,0 +1,195 @@
+    pub fn data_globs(&self) -> Option<Vec<String>> {
+        match self.pyext.as_ref() {
+            Some(exts) => {
+                let globs = exts
+                    .iter()
+                    .filter(|&ext| ext != "py" && ext != "pyi")
+                    .map(|ext| format!("\"**/*.{}\"", ext))
+                    .collect_vec();
+                if globs.is_empty() {
+                    None
+                } else {
+                    Some(globs)
+                }
+            }
+
+            None => None,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct Kernel {
+    pub backend: Backend,
+    pub cuda_capabilities: Option<Vec<String>>,
+    pub rocm_archs: Option<Vec<String>>,
+    pub depends: Vec<Dependencies>,
+    pub include: Option<Vec<String>>,
+    pub src: Vec<String>,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd, Serialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub enum Backend {
+    Cuda,
+    Rocm,
+}
+
+impl Display for Backend {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Backend::Cuda => write!(f, "cuda"),
+            Backend::Rocm => write!(f, "rocm"),
+        }
+    }
+}
+
+impl FromStr for Backend {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "cuda" => Ok(Backend::Cuda),
+            "rocm" => Ok(Backend::Rocm),
+            _ => Err(format!("Unknown backend: {}", s)),
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "lowercase")]
+pub enum Dependencies {
+    #[serde[rename = "cutlass_2_10"]]
+    Cutlass2_10,
+    #[serde[rename = "cutlass_3_5"]]
+    Cutlass3_5,
+    #[serde[rename = "cutlass_3_6"]]
+    Cutlass3_6,
+    #[serde[rename = "cutlass_3_8"]]
+    Cutlass3_8,
+    Torch,
+}
+
+impl From<v1::Build> for Build {
+    fn from(build: v1::Build) -> Self {
+        let universal = build
+            .torch
+            .as_ref()
+            .map(|torch| torch.universal)
+            .unwrap_or(false);
+        Self {
+            general: General::from(build.general, universal),
+            torch: build.torch.map(Into::into),
+            kernels: convert_kernels(build.kernels),
+        }
+    }
+}
+
+impl General {
+    fn from(general: v1::General, universal: bool) -> Self {
+        Self {
+            name: general.name,
+            universal,
+        }
+    }
+}
+
+fn convert_kernels(v1_kernels: HashMap<String, v1::Kernel>) -> HashMap<String, Kernel> {
+    let mut kernels = HashMap::new();
+
+    for (name, kernel) in v1_kernels {
+        if kernel.language == Language::CudaHipify {
+            kernels.insert(
+                // We need to add an affix to avoid confflict with the CUDA kernel.
```

Comment:
I think a collision is possible if the user has a hard coded `{name}_rocm` not sure if we want to check for uniqueness?